### PR TITLE
Source Hubspot: hotfix client tests

### PR DESCRIPTION
**Pull request by keu**
## What
It seems that Cloudflare enforced minimal length for hapikey parameter, this causes our custom integration test to fail.

## How
Improved fallback and health_check logic as well.

## Pre-merge Checklist
- [ ] *Run integration tests*
- [ ] *Publish Docker images*

## Recommended reading order
1. `test.java`
1. `component.ts`
1. the rest

### CODE REVIEW

File: airbyte-integrations/connectors/source-hubspot/source_hubspot/client.py [hunk 1]
```diff
@@ -102,7 +102,7 @@ def health_check(self) -> Tuple[bool, str]:
 
         try:
             _ = self._apis["contacts"].properties
-        except HubspotError as error:
+        except HTTPError as error:
```
Comment (arhip11) [inline airbyte-integrations/connectors/source-hubspot/source_hubspot/client.py:105]: To discuss: what's the purpose of HubspotError hierarchy then?

Comment (keu) [reply airbyte-integrations/connectors/source-hubspot/source_hubspot/client.py:105]: to distinct API errors from others

File: airbyte-integrations/connectors/source-hubspot/integration_tests/client_test.py [hunk 2]
```diff
@@ -29,7 +29,7 @@
 
 @pytest.fixture(name="wrong_credentials")
 def wrong_credentials_fixture():
-    return {"api_key": "wrong_key"}
+    return {"api_key": "wrongkey-key1-key2-key3-wrongkey1234"}
```
[the remark this answers is not in the document]
Comment (keu) [reply airbyte-integrations/connectors/source-hubspot/integration_tests/client_test.py:32]: no, it is a combination of both, I made a key that looks like real, but still far from it

**Review by sherifnada — APPROVED** [review]

**Review by arhip11 — APPROVED** [review]

**Pull request merged**